src/input/iterator.rs

```diff
@@ -117,16 +117,10 @@ impl<'data> IterableValidationChecks<'data> {
         input: &'data I,
     ) -> ValResult<'data, ()> {
         if max_length < current_length {
-            return Err(ValError::new(
-                ErrorType::TooLong {
-                    field_type: self.field_type.to_string(),
-                    max_length,
-                    actual_length: current_length,
-                },
-                input,
-            ));
+            Ok(())
```

Comment:
What makes no sense is that this code path never gets hits in the benchmarks, yet it makes a big difference in them!